File: FitToFit/FitToFit.Model/Rezervacije.cs
using Microsoft.EntityFrameworkCore.Storage;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FitToFit.Model
{
    public class Rezervacije
    {
        public int RezervacijaId { get; set; }

        public DateTime Datum { get; set; }

        public int KorisnikId { get; set; }

        public int ClanarinaId { get; set; }

        public string? StateMachine { get; set; }

        public decimal? Iznos { get; set; }

        public DateTime? DatumIsteka { get; set; }

        //public virtual Korisnici Korisnik { get; set; } = null!; // ???

        public virtual ICollection<RezervacijaStavke> RezervacijaStavkes { get; set; } = new List<RezervacijaStavke>();
    }
}

File: FitToFit/UI/fittofit_admin/lib/models/rezervacije.dart
import 'package:fittofit_admin/models/rezervacijaStavke.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacije.g.dart';

@JsonSerializable()
class Rezervacije {
  int rezervacijaId;
  DateTime datum;
  String? stateMachine;
  double? iznos;
  int korisnikId;
  int clanarinaId;
  String? brojTransakcije;
  List<RezervacijaStavke>? rezervacijaStavkes = [];

  Rezervacije(
      {required this.rezervacijaId,
      required this.datum,
      this.stateMachine,
      this.iznos,
      required this.korisnikId,
      required this.clanarinaId,
      this.brojTransakcije,
      this.rezervacijaStavkes});

  factory Rezervacije.fromJson(Map<String, dynamic> json) =>
      _$RezervacijeFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijeToJson(this);
}
